master/internal/stream/system.go

```diff
@@ -218,37 +277,79 @@ func (ps PublisherSet) Websocket(socket *websocket.Conn, c echo.Context) error {
 
 		// any modifications to our subscriptions?
 		for _, mod := range mods {
-			temp, err := ss.SubscriptionMod(mod, ctx)
+			temp, err := ss.SubscriptionMod(ctx, mod)
 			if err != nil {
 				return errors.Wrapf(err, "error modifying subscriptions")
 			}
 			msgs = append(msgs, temp...)
-			// TODO: also append a sync message (or one sync per SubscriptionModMsg)
+			// XXX: also append a sync message (or one sync per SubscriptionModMsg)
 		}
 
 		// write msgs to the websocket
 		err = writeAll(socket, msgs)
 		if err != nil {
-			// TODO: don't log broken pipe errors.
+			// XXX: don't log broken pipe errors.
 			if err != nil {
 				return errors.Wrapf(err, "error writing to socket")
 			}
 		}
 	}
+}
 
-	return nil
+func (ps *PublisherSet) permissionChangeLoop(ctx context.Context) {
+	permListener, err := AuthZProvider.Get().GetPermissionChangeListener()
+	switch {
+	case err != nil:
+		log.Errorf("error occurred while getting permission change listener: %s", err)
+		fallthrough
+	case permListener == nil:
+		// no need to loop or cleanup if there's no listener
+		return
+	}
```

Comment:
This switch statement feels weird, is there a better approach for handling

Reply:
I agree, it's a weird switch statement.  Did go force you to write a switch statement?

I think a more natural expression would be:


```
func(...){
  l, err := GetListener()
  if err != nil {
    log.Error(...)
     return nil
  }
  if l == nil {
    // no listener means we don't have permissions configured at all
    return nil
  }
  ...
}
```

squeezing it into a switch statement feels odd.

Also, ignoring errors feels odd.   What if there's a legitimate error here, do we just silently decide, 'oops you don't get rbac safety today'?

Reply:
I've had the linter scream too many times at me about if/switch cases, I like the two if statement approach much more!

Yea, and I agree ignoring the errors here doesn't make sense, it should cause `ps.Start()` to fail; no as far as how/if it recovers is another question.

Reply:
alternate approach could be to pass the PublisherSet to rbac, and just have the rbac implementation call ps.Reset() or whatever.

Not sure what's best, that's just an idea.